nemoguardrails/rails/llm/llmrails.py

```
@@ -118,17 +118,34 @@
 process_events_semaphore = asyncio.Semaphore(1)
 
 
+def _wrap_legacy_llm(llm):
+    try:
+        from nemoguardrails.integrations.langchain.llm_adapter import LangChainLLMAdapter
+    except ImportError:
+        raise TypeError(
+            "Passing a raw LangChain LLM requires langchain to be installed. "
+            "Either install langchain or pass an LLMModel instance."
+        )
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)